diff -burp freetype-2.4.10/include/freetype/config/ftoption.h freetype-2.4.10-patched/include/freetype/config/ftoption.h
--- freetype-2.4.10/include/freetype/config/ftoption.h	2012-06-14 01:35:58.000000000 -0400
+++ freetype-2.4.10-patched/include/freetype/config/ftoption.h	2012-08-15 13:19:43.000000000 -0400
@@ -148,7 +148,7 @@ FT_BEGIN_HEADER
   /*                                                                       */
   /*   Define this macro if you want to enable this `feature'.             */
   /*                                                                       */
-#define FT_CONFIG_OPTION_USE_LZW
+/*#define FT_CONFIG_OPTION_USE_LZW*/
 
 
   /*************************************************************************/
@@ -163,7 +163,7 @@ FT_BEGIN_HEADER
   /*   Define this macro if you want to enable this `feature'.  See also   */
   /*   the macro FT_CONFIG_OPTION_SYSTEM_ZLIB below.                       */
   /*                                                                       */
-#define FT_CONFIG_OPTION_USE_ZLIB
+/*#define FT_CONFIG_OPTION_USE_ZLIB*/
 
 
   /*************************************************************************/
@@ -278,7 +278,7 @@ FT_BEGIN_HEADER
   /*   You would normally undefine this configuration macro when building  */
   /*   a version of FreeType that doesn't contain a Type 1 or CFF driver.  */
   /*                                                                       */
-#define FT_CONFIG_OPTION_POSTSCRIPT_NAMES
+/*#define FT_CONFIG_OPTION_POSTSCRIPT_NAMES*/
 
 
   /*************************************************************************/
@@ -296,7 +296,7 @@ FT_BEGIN_HEADER
   /*   able to synthesize a Unicode charmap out of the glyphs found in the */
   /*   fonts.                                                              */
   /*                                                                       */
-#define FT_CONFIG_OPTION_ADOBE_GLYPH_LIST
+/*#define FT_CONFIG_OPTION_ADOBE_GLYPH_LIST*/
 
 
   /*************************************************************************/
@@ -309,7 +309,7 @@ FT_BEGIN_HEADER
   /*                                                                       */
   /*   Note that the `FOND' resource isn't checked.                        */
   /*                                                                       */
-#define FT_CONFIG_OPTION_MAC_FONTS
+/*#define FT_CONFIG_OPTION_MAC_FONTS*/
 
 
   /*************************************************************************/
@@ -343,7 +343,7 @@ FT_BEGIN_HEADER
   /* supply font data incrementally as the document is parsed, such        */
   /* as the Ghostscript interpreter for the PostScript language.           */
   /*                                                                       */
-#define FT_CONFIG_OPTION_INCREMENTAL
+/*#define FT_CONFIG_OPTION_INCREMENTAL*/
 
 
   /*************************************************************************/
@@ -365,7 +365,7 @@ FT_BEGIN_HEADER
   /*   The maximum number of modules that can be registered in a single    */
   /*   FreeType library object.  32 is the default.                        */
   /*                                                                       */
-#define FT_MAX_MODULES  32
+#define FT_MAX_MODULES  8
 
 
   /*************************************************************************/
@@ -502,7 +502,7 @@ FT_BEGIN_HEADER
   /*                                                                       */
   /* (By default, the module uses `PSNames' to extract glyph names.)       */
   /*                                                                       */
-#define TT_CONFIG_OPTION_POSTSCRIPT_NAMES
+/*#define TT_CONFIG_OPTION_POSTSCRIPT_NAMES*/
 
 
   /*************************************************************************/
@@ -516,7 +516,7 @@ FT_BEGIN_HEADER
   /* Accessing SFNT names is done through the functions declared in        */
   /* `freetype/ftsnames.h'.                                                */
   /*                                                                       */
-#define TT_CONFIG_OPTION_SFNT_NAMES
+/*#define TT_CONFIG_OPTION_SFNT_NAMES*/
 
 
   /*************************************************************************/
@@ -647,7 +647,7 @@ FT_BEGIN_HEADER
   /* and avar tables).  This has many similarities to Type 1 Multiple      */
   /* Masters support.                                                      */
   /*                                                                       */
-#define TT_CONFIG_OPTION_GX_VAR_SUPPORT
+/*#define TT_CONFIG_OPTION_GX_VAR_SUPPORT*/
 
 
   /*************************************************************************/
@@ -655,7 +655,7 @@ FT_BEGIN_HEADER
   /* Define TT_CONFIG_OPTION_BDF if you want to include support for        */
   /* an embedded `BDF ' table within SFNT-based bitmap formats.            */
   /*                                                                       */
-#define TT_CONFIG_OPTION_BDF
+/*#define TT_CONFIG_OPTION_BDF*/
 
 
   /*************************************************************************/
@@ -733,7 +733,7 @@ FT_BEGIN_HEADER
   /*                                                                       */
   /* Compile autofit module with Indic script support.                     */
   /*                                                                       */
-#define AF_CONFIG_OPTION_INDIC
+/*#define AF_CONFIG_OPTION_INDIC*/
 
   /*************************************************************************/
   /*                                                                       */
@@ -761,7 +761,7 @@ FT_BEGIN_HEADER
    * is recommended to disable the macro since it reduces the library's code
    * size and activates a few memory-saving optimizations as well.
    */
-#define FT_CONFIG_OPTION_OLD_INTERNALS
+/*#define FT_CONFIG_OPTION_OLD_INTERNALS*/
 
 
   /*
diff -burp freetype-2.4.10/modules.cfg freetype-2.4.10-patched/modules.cfg
--- freetype-2.4.10/modules.cfg	2011-11-26 07:19:10.000000000 -0500
+++ freetype-2.4.10-patched/modules.cfg	2012-08-15 13:15:25.000000000 -0400
@@ -37,35 +37,35 @@ FONT_MODULES += truetype
 # PostScript Type 1 font driver.
 #
 # This driver needs the `psaux', `pshinter', and `psnames' modules.
-FONT_MODULES += type1
+#FONT_MODULES += type1
 
 # CFF/OpenType font driver.
 #
 # This driver needs the `sfnt', `pshinter', and `psnames' modules.
-FONT_MODULES += cff
+#FONT_MODULES += cff
 
 # Type 1 CID-keyed font driver.
 #
 # This driver needs the `psaux', `pshinter', and `psnames' modules.
-FONT_MODULES += cid
+#FONT_MODULES += cid
 
 # PFR/TrueDoc font driver.  See optional extension ftpfr.c below also.
-FONT_MODULES += pfr
+#FONT_MODULES += pfr
 
 # PostScript Type 42 font driver.
 #
-# This driver needs the `truetype' and `psaux' modules.
-FONT_MODULES += type42
+# This driver needs the `truetype' module.
+#FONT_MODULES += type42
 
 # Windows FONT/FNT font driver.  See optional extension ftwinfnt.c below
 # also.
-FONT_MODULES += winfonts
+#FONT_MODULES += winfonts
 
 # PCF font driver.
-FONT_MODULES += pcf
+#FONT_MODULES += pcf
 
 # BDF font driver.  See optional extension ftbdf.c below also.
-FONT_MODULES += bdf
+#FONT_MODULES += bdf
 
 # SFNT files support.  If used without `truetype' or `cff', it supports
 # bitmap-only fonts within an SFNT wrapper.
@@ -79,10 +79,10 @@ FONT_MODULES += sfnt
 ####
 
 # FreeType's auto hinter.
-HINTING_MODULES += autofit
+#HINTING_MODULES += autofit
 
 # PostScript hinter.
-HINTING_MODULES += pshinter
+#HINTING_MODULES += pshinter
 
 # The TrueType hinting engine doesn't have a module of its own but is
 # controlled in file include/freetype/config/ftoption.h
@@ -94,7 +94,7 @@ HINTING_MODULES += pshinter
 ####
 
 # Monochrome rasterizer.
-RASTER_MODULES += raster
+#RASTER_MODULES += raster
 
 # Anti-aliasing rasterizer.
 RASTER_MODULES += smooth
@@ -107,7 +107,7 @@ RASTER_MODULES += smooth
 # FreeType's cache sub-system (quite stable but still in beta -- this means
 # that its public API is subject to change if necessary).  See
 # include/freetype/ftcache.h.  Needs ftglyph.c.
-AUX_MODULES += cache
+#AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs ftgxval.c below.
 # AUX_MODULES += gxvalid
@@ -115,17 +115,17 @@ AUX_MODULES += cache
 # Support for streams compressed with gzip (files with suffix .gz).
 #
 # See include/freetype/ftgzip.h for the API.
-AUX_MODULES += gzip
+#AUX_MODULES += gzip
 
 # Support for streams compressed with LZW (files with suffix .Z).
 #
 # See include/freetype/ftlzw.h for the API.
-AUX_MODULES += lzw
+#AUX_MODULES += lzw
 
 # Support for streams compressed with bzip2 (files with suffix .bz2).
 #
 # See include/freetype/ftbzip2.h for the API.
-AUX_MODULES += bzip2
+#AUX_MODULES += bzip2
 
 # OpenType table validation.  Needs ftotval.c below.
 #
@@ -134,7 +134,7 @@ AUX_MODULES += bzip2
 # Auxiliary PostScript driver component to share common code.
 #
 # This module depends on `psnames'.
-AUX_MODULES += psaux
+#AUX_MODULES += psaux
 
 # Support for PostScript glyph names.
 #
@@ -150,95 +150,95 @@ AUX_MODULES += psnames
 # Exact bounding box calculation.
 #
 # See include/freetype/ftbbox.h for the API.
-BASE_EXTENSIONS += ftbbox.c
+#BASE_EXTENSIONS += ftbbox.c
 
 # Access BDF-specific strings.  Needs BDF font driver.
 #
 # See include/freetype/ftbdf.h for the API.
-BASE_EXTENSIONS += ftbdf.c
+#BASE_EXTENSIONS += ftbdf.c
 
 # Utility functions for converting 1bpp, 2bpp, 4bpp, and 8bpp bitmaps into
 # 8bpp format, and for emboldening of bitmap glyphs.
 #
 # See include/freetype/ftbitmap.h for the API.
-BASE_EXTENSIONS += ftbitmap.c
+#BASE_EXTENSIONS += ftbitmap.c
 
 # Access CID font information.
 #
 # See include/freetype/ftcid.h for the API.
-BASE_EXTENSIONS += ftcid.c
+#BASE_EXTENSIONS += ftcid.c
 
 # Access FSType information.  Needs fttype1.c.
 #
 # See include/freetype/freetype.h for the API.
-BASE_EXTENSIONS += ftfstype.c
+#BASE_EXTENSIONS += ftfstype.c
 
 # Support for GASP table queries.
 #
 # See include/freetype/ftgasp.h for the API.
-BASE_EXTENSIONS += ftgasp.c
+#BASE_EXTENSIONS += ftgasp.c
 
 # Convenience functions to handle glyphs.  Needs ftbitmap.c.
 #
 # See include/freetype/ftglyph.h for the API.
-BASE_EXTENSIONS += ftglyph.c
+#BASE_EXTENSIONS += ftglyph.c
 
 # Interface for gxvalid module.
 #
 # See include/freetype/ftgxval.h for the API.
-BASE_EXTENSIONS += ftgxval.c
+#BASE_EXTENSIONS += ftgxval.c
 
 # Support for LCD color filtering of subpixel bitmaps.
 #
 # See include/freetype/ftlcdfil.h for the API.
-BASE_EXTENSIONS += ftlcdfil.c
+#BASE_EXTENSIONS += ftlcdfil.c
 
 # Multiple Master font interface.
 #
 # See include/freetype/ftmm.h for the API.
-BASE_EXTENSIONS += ftmm.c
+#BASE_EXTENSIONS += ftmm.c
 
 # Interface for otvalid module.
 #
 # See include/freetype/ftotval.h for the API.
-BASE_EXTENSIONS += ftotval.c
+#BASE_EXTENSIONS += ftotval.c
 
 # Support for FT_Face_CheckTrueTypePatents.
 #
 # See include/freetype/freetype.h for the API.
-BASE_EXTENSIONS += ftpatent.c
+#BASE_EXTENSIONS += ftpatent.c
 
 # Interface for accessing PFR-specific data.  Needs PFR font driver.
 #
 # See include/freetype/ftpfr.h for the API.
-BASE_EXTENSIONS += ftpfr.c
+#BASE_EXTENSIONS += ftpfr.c
 
 # Path stroker.  Needs ftglyph.c.
 #
 # See include/freetype/ftstroke.h for the API.
-BASE_EXTENSIONS += ftstroke.c
+#BASE_EXTENSIONS += ftstroke.c
 
 # Support for synthetic embolding and slanting of fonts.  Needs ftbitmap.c.
 #
 # See include/freetype/ftsynth.h for the API.
-BASE_EXTENSIONS += ftsynth.c
+#BASE_EXTENSIONS += ftsynth.c
 
 # Interface to access data specific to PostScript Type 1 and Type 2 (CFF)
 # fonts.
 #
 # See include/freetype/t1tables.h for the API.
-BASE_EXTENSIONS += fttype1.c
+#BASE_EXTENSIONS += fttype1.c
 
 # Interface for accessing data specific to Windows FNT files.  Needs winfnt
 # driver.
 #
 # See include/freetype/ftwinfnt.h for the API.
-BASE_EXTENSIONS += ftwinfnt.c
+#BASE_EXTENSIONS += ftwinfnt.c
 
 # Support functions for X11.
 #
 # See include/freetype/ftxf86.h for the API.
-BASE_EXTENSIONS += ftxf86.c
+#BASE_EXTENSIONS += ftxf86.c
 
 ####
 #### The components `ftsystem.c' (for memory allocation and stream I/O
