diff --git a/build.gradle b/build1.gradle
index 9d920e9..e9df219 100644
--- a/build.gradle
+++ b/build1.gradle
@@ -26,22 +26,12 @@ android {
         archivesBaseName = METADATA_PROJECT
     }
 
-    // set code signing variables in ~/.gradle/gradle.properties
-    signingConfigs {
-        release {
-            storeFile file(ANDROID_KEY_STORE)
-            storePassword ANDROID_KEY_STORE_PASSWORD
-            keyAlias ANDROID_KEY_ALIAS
-            keyPassword ANDROID_KEY_PASSWORD
-        }
-    }
-
     buildTypes {
         release {
             shrinkResources true
             minifyEnabled true
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard.cfg'
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.debug
         }
     }
 
